Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm

```diff
@@ -295,12 +295,12 @@ static void replaceRichContentWithAttachments(Frame& frame, DocumentFragment& fr
     // FIXME: Handle resources in subframe archives.
     HashMap<AtomString, Ref<ArchiveResource>> urlToResourceMap;
     for (auto& subresource : subresources)
-        urlToResourceMap.set(subresource->url().string(), subresource.copyRef());
+        urlToResourceMap.set(AtomString { subresource->url().string() }, subresource.copyRef());
 
     Vector<SerializedAttachmentData> serializedAttachmentData;
     for (auto& attachment : descendantsOfType<HTMLAttachmentElement>(fragment)) {
         auto resourceURL = HTMLAttachmentElement::archiveResourceURL(attachment.uniqueIdentifier());
-        auto resourceEntry = urlToResourceMap.find(resourceURL.string());
+        auto resourceEntry = urlToResourceMap.find(AtomString { resourceURL.string() });
```

Comment:
Given this is a find of an existing `AtomString` key, it can use an “existing atom string” code path to avoid waste fully allocating any temporary strings. We have that in `StringView`, not sure if we have the same for `String`.